### ISSUE TITLE
1.5-vibe-voice-lora!

### ISSUE THREAD

**Issue by al-swaiti** [voicepowered-ai/VibeVoice-finetuning#13]
I don’t know! There was no improvement! I prepared  dataset  https://huggingface.co/datasets/ABDALLALSWAITI/vibevoice-arabic-dataset

https://wandb.ai/abedalswaity7/huggingface/runs/x0ar4ul7/logs

, then trained for 5 epochs as follows, but  the loras not effect the model output ! .
 I tried all checkpoints and there was no improvement in pronunciation! 
Did I do something wrong

!python -m src.finetune_vibevoice_lora \
  --model_name_or_path vibevoice/VibeVoice-1.5B  \
  --processor_name_or_path src/vibevoice/processor \
  --dataset_name ABDALLALSWAITI/vibevoice-arabic-dataset \
  --text_column_name text \
  --audio_column_name audio \
  --output_dir ./output_arabic_lora \
  --per_device_train_batch_size 8 \
  --gradient_accumulation_steps 16 \
  --learning_rate 2.5e-5 \
  --num_train_epochs 5 \
  --logging_steps 10 \
  --save_steps 100 \
  --eval_steps 100 \
  --report_to wandb \
  --remove_unused_columns False \
  --bf16 True \
  --do_train \
  --gradient_clipping \
  --gradient_checkpointing False \
  --ddpm_batch_mul 4 \
  --diffusion_loss_weight 1.4 \
  --train_diffusion_head True \
  --ce_loss_weight 0.04 \
  --voice_prompt_drop_rate 0.2 \
  --lora_target_modules q_proj,k_proj,v_proj,o_proj,gate_proj,up_proj,down_proj \
  --lr_scheduler_type cosine \
  --warmup_ratio 0.03 \
  --max_grad_norm 0.8 \
  --dataloader_num_workers 4 \
  --dataloader_pin_memory True \
  --seed 42

- the checkpoints https://huggingface.co/ABDALLALSWAITI/vibevoice-arabic-1.5/tree/main

**Comment by jpgallegoar-vpai** [comment]
Make sure the LORAs are being applied on inference, oftentimes there are error messages in the console logs. Which repo did you use to test it?

**Comment by al-swaiti** [comment]
https://github.com/Enemyx-net/VibeVoice-ComfyUI
<details>
<summary>comfyui-logs</summary>
[VibeVoice] LoRA configured: vibevoice-arabic-1.5-z (/home/abdallah/Desktop/ComfyUI/models/vibevoice/loras/vibevoice-arabic-1.5-z)
[VibeVoice] Additional LoRA components found: diffusion_head, acoustic_connector, semantic_connector
[VibeVoice] LoRA components enabled: LLM (strength: 1.0), Diffusion Head, Acoustic Connector, Semantic Connector
[VibeVoice] Using LoRA from: /home/abdallah/Desktop/ComfyUI/models/vibevoice/loras/vibevoice-arabic-1.5-z
APEX FusedRMSNorm not available, using native implementation
[VibeVoice] Using embedded VibeVoice from /home/abdallah/Desktop/ComfyUI/custom_nodes/VibeVoice-ComfyUI/vvembed
[VibeVoice] Found local model snapshot: /home/abdallah/Desktop/ComfyUI/models/vibevoice/models--microsoft--VibeVoice-1.5B/snapshots/1904eae38036e9c780d28e27990c27748984eafe
[VibeVoice] Using auto attention implementation selection
[VibeVoice] Loading model from local snapshot: /home/abdallah/Desktop/ComfyUI/models/vibevoice/models--microsoft--VibeVoice-1.5B/snapshots/1904eae38036e9c780d28e27990c27748984eafe
Loading checkpoint shards:  33%|██    | 1/3 [00:01<00:02,  1.30s/it]FETCH ComfyRegistry Data: 70/98
Loading checkpoint shards: 100%|██████| 3/3 [00:03<00:00,  1.15s/it]
No preprocessor_config.json found at /home/abdallah/Desktop/ComfyUI/models/vibevoice/models--microsoft--VibeVoice-1.5B/snapshots/1904eae38036e9c780d28e27990c27748984eafe, using defaults
Loading tokenizer from Qwen/Qwen2.5-1.5B
[VibeVoice] Starting LoRA application from path: /home/abdallah/Desktop/ComfyUI/models/vibevoice/loras/vibevoice-arabic-1.5-z
[VibeVoice] LoRA component flags - LLM: True, Diffusion: True, Acoustic: True, Semantic: True
[VibeVoice] Applying LLM LoRA adapter from: /home/abdallah/Desktop/ComfyUI/models/vibevoice/loras/vibevoice-arabic-1.5-z
[VibeVoice] Failed to apply LLM LoRA: 'Qwen2Model' object has no attribute 'prepare_inputs_for_generation'
[VibeVoice] Found diffusion_head directory at: /home/abdallah/Desktop/ComfyUI/models/vibevoice/loras/vibevoice-arabic-1.5-z/diffusion_head
[VibeVoice] Found prediction_head module in model
[VibeVoice] Current model prediction_head hidden size: 1536
[VibeVoice] LoRA diffusion head hidden size: 1536
[VibeVoice] Loading safetensor from: /home/abdallah/Desktop/ComfyUI/models/vibevoice/loras/vibevoice-arabic-1.5-z/diffusion_head/model.safetensors
[VibeVoice] Loaded state dict with 26 keys
[VibeVoice] Module device: cuda:0, dtype: torch.bfloat16
[VibeVoice] Successfully loaded state dict into module
[VibeVoice] Diffusion head LoRA loaded successfully into prediction_head
[VibeVoice] Loading pytorch model from: /home/abdallah/Desktop/ComfyUI/models/vibevoice/loras/vibevoice-arabic-1.5-z/acoustic_connector/pytorch_model.bin
[VibeVoice] Loaded state dict with 5 keys
[VibeVoice] Successfully loaded pytorch model into module
[VibeVoice] Acoustic connector LoRA loaded
[VibeVoice] Loading pytorch model from: /home/abdallah/Desktop/ComfyUI/models/vibevoice/loras/vibevoice-arabic-1.5-z/semantic_connector/pytorch_model.bin
[VibeVoice] Loaded state dict with 5 keys
[VibeVoice] Successfully loaded pytorch model into module
[VibeVoice] Semantic connector LoRA loaded
[VibeVoice] LoRA application completed
[VibeVoice] Processing text segment 1 (16 words)
[VibeVoice] Starting audio generation with 20 diffusion steps...
[VibeVoice] Generating audio with 20 diffusion steps...
[VibeVoice] Note: Progress bar shows max possible tokens, not actual needed (~45 estimated)
[VibeVoice] The generation will stop automatically when audio is complete
Generating (active: 1/1):   0%|    | 6/1498 [00:02<05:28,  4.54it/s]FETCH ComfyRegistry Data: 75/98
Generating (active: 1/1):   4%|   | 54/1498 [00:06<02:13, 10.78it/s]FETCH ComfyRegistry Data: 80/98
[VibeVoice] Concatenating 1 audio segments (including pauses)...    
[VibeVoice] Successfully generated audio with 1 segments

</details>

**Comment by al-swaiti** [comment]
also i tried this # WITH LoRA (built-in Alice voice)
os.makedirs('/root/outputs/builtin_with_lora', exist_ok=True)
!python /root/VibeVoice/demo/inference_from_file.py \
    --model_path /root/models/VibeVoice-1.5B\
    --txt_path /root/test.txt \
    --speaker_names Alice Frank \
    --checkpoint_path /root/models/vibevoice-arabic-1.5-z \
    --output_dir /root/outputs/builtin_with_lora  same! the result has no enhacment

**Comment by jpgallegoar-vpai** [comment]
Look at the log, it says "Failed to apply LLM LoRA: 'Qwen2Model' object has no attribute 'prepare_inputs_for_generation'"

Do you have transformers==4.51.3 in the comfy env?

**Comment by al-swaiti** [comment]
yeah thanks thats work now

**Comment by ILG2021** [comment]
I encountered the same problem, I don't have any errors, but the result timbre is not the lora. I have finetuned with a dataset about 100 hours. I found I made a mistake that I edit the text label to "speaker 0: xxxx", not "Speaker 0: xxxx", may it affect the generation?
@jpgallegoar-vpai

**Comment by jpgallegoar-vpai** [comment]
> I encountered the same problem, I don't have any errors, but the result timbre is not the lora. I have finetuned with a dataset about 100 hours. I found I made a mistake that I edit the text label to "speaker 0: xxxx", not "Speaker 0: xxxx", may it affect the generation? [@jpgallegoar-vpai](https://github.com/jpgallegoar-vpai)

yes, it's possible, the model expects that format exactly
